llvm/test/CodeGen/AMDGPU/amdgpu-attributor-no-agpr.ll

```diff
@@ -231,7 +231,19 @@ define amdgpu_kernel void @indirect_calls_none_agpr(i1 %cond) {
 ; CHECK-LABEL: define amdgpu_kernel void @indirect_calls_none_agpr(
 ; CHECK-SAME: i1 [[COND:%.*]]) #[[ATTR0]] {
 ; CHECK-NEXT:    [[FPTR:%.*]] = select i1 [[COND]], ptr @empty, ptr @also_empty
-; CHECK-NEXT:    call void [[FPTR]]()
+; CHECK-NEXT:    [[TMP1:%.*]] = icmp eq ptr [[FPTR]], @also_empty
+; CHECK-NEXT:    br i1 [[TMP1]], label [[TMP2:%.*]], label [[TMP3:%.*]]
+; CHECK:       2:
+; CHECK-NEXT:    call void @also_empty()
+; CHECK-NEXT:    br label [[TMP6:%.*]]
+; CHECK:       3:
+; CHECK-NEXT:    br i1 true, label [[TMP4:%.*]], label [[TMP5:%.*]]
+; CHECK:       4:
+; CHECK-NEXT:    call void @empty()
+; CHECK-NEXT:    br label [[TMP6]]
+; CHECK:       5:
+; CHECK-NEXT:    unreachable
+; CHECK:       6:
```

Comment:
This looks like an accidental test change, and it's not obvious to me it's better? I also thought AAIndirectCallInfo would just allow analysis of indirect calls, but this is doing the indirect call to conditional call promotion?

Reply:
It is not an accident. `AAIndirectCallInfo` will do both analysis and manifest. This change is expected (for now) because it replaces an indirect function call with two direct function calls under condition. This is assumed to have better performance. The reason it is for now is, in https://github.com/llvm/llvm-project/pull/100954 we add uniform check, and this change doesn't conform with it.

Reply:
I understand your point, but AAs don't run in isolation. Here we enable a new AA so some tests need to be updated. I'd argue this doesn't defect the purpose of the test. Also, in #100954 this is reverted.

This patch primes #100953 to help to demonstrate how the changes in #100953 works; otherwise I'd just enable that in #100954.

Reply:
I mean there should be a new test, specifically to test this. The test here should be changed so it remains an indirect call

Reply:
The default heuristic will only expand it to an if-cascade if there are 2? potential callees. Make it more than that and it won't expand, or change the default number via command line args.